R/plot.R

```diff
@@ -68,57 +68,196 @@ fldts2df <- function(fldts, griddata)
+#' @param palettestr String with the name of the ggplot2 color palette to use.
+#' Defaults to 'RdYlBu'. Options are Diverging: 'BrBG', 'PiYG', 'PRGn', 'PuOr',
+#' 'RdBu', 'RdGy', 'RdYlBu', 'RdYlGn', 'Spectral'; Qualitative: 'Accent',
+#' 'Dark2', 'Paired', 'Pastel1', 'Pastel2', 'Set1', 'Set2', 'Set3';
+#' Sequential: 'Blues', 'BuGn', 'BuPu', 'GnBu', 'Greens', 'Greys', 'Oranges',
+#' 'OrRd', 'PuBu', 'PuBuGn', 'PuRd', 'Purples', 'RdPu', 'Reds', 'YlGn',
+#' 'YlGnBu', 'YlOrBr', 'YlOrRd'.
+#' @param palettedir Numerical value, only accepts 1 or -1, determining the
+#' order of the color palette used. Defaults to -1.
+#' @export
+plot_field <- function(fld, griddata, nb=6, minval=-3.5, maxval=3.5, legendstr="Temperature (K)", palettestr = 'RdYlBu', palettedir = -1 )
+{
+
+        # reshape data to be easier to plot
+        tdf <- fld2df(fld, griddata)
+
+        # get outlines of countries to add to map to make them easier to interpret
+        # Do tweak the easter tip of Russia to get the map to break evenly across
+        # longitudes.
+        countries <- ggplot2::map_data('world') %>% mutate(value=0)
+        maxgroup <- max(countries$group)
+
+        countries %>%
+            mutate(group = if_else(long > 180, maxgroup + 1, group),
+                   long = if_else(long > 180, long - 360, long)) ->
+            countries
+
+        ## TODO: make these options a little more customizable
+        if(nb < 2) {
+            ggplot2::ggplot(tdf, aes(x=lon, y = lat, fill = value)) +
+                ggplot2::geom_raster() +
+                ggplot2::scale_fill_distiller(palette=palettestr, direction=palettedir,
+                                              limits=c(minval, maxval), oob=scales::squish,
+                                              guide=ggplot2::guide_colorbar(title=legendstr, title.position='top')) +
+                ggplot2::scale_x_continuous(expand=c(0,0)) +
+                ggplot2::scale_y_continuous(expand=c(0,0)) +
+                ggplot2::theme(axis.title=element_blank(),
+                              axis.text=element_blank(),
+                              axis.ticks=element_blank(),
+                              legend.position = 'bottom') +
+                # black outlines of countries
+                ggplot2::geom_path(data = countries,
+                                   aes(x = long, y = lat, group = group),
+                                   size = 0.25, color = 'black')
+
+            # # Legacy code using gcammaptools
+            # gcammaptools::plot_GCAM_grid(tdf, col='value', extent=gcammaptools::EXTENT_WORLD,
+            #                              legend=TRUE) +
+            #     ggplot2::scale_fill_distiller(palette=palettestr, direction=palettedir,
+            #                                   limits=c(minval, maxval), oob=scales::squish,
+            #                                   guide=ggplot2::guide_colorbar(title=legendstr, title.position='top'))
+
```

Comment:
Delete the legacy code, it isn't going to be used any more and if some one really wants it they can look at older versions of fldgen